core/src/include/Xios.hpp

```diff
@@ -0,0 +1,84 @@
+/*!
+ * @file    Xios.hpp
+ * @author  Tom Meltzer <tdm39@cam.ac.uk>
+ * @date    Fri 23 Feb 13:43:16 GMT 2024
+ * @brief   XIOS interface header
+ * @details
+ *
+ * Header file for XIOS interface
+ *
+ */
+#ifndef SRC_INCLUDE_XIOS_HPP
+#define SRC_INCLUDE_XIOS_HPP
+
+#include "date.hpp"
+#if USE_XIOS
+
+#include "Configured.hpp"
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/format.hpp>
+#include <boost/format/group.hpp>
+#include <include/xios_c_interface.hpp>
+#include <mpi.h>
+
+namespace Nextsim {
+
+class Xios : public Configured<Xios> {
+public:
+    Xios();
+    ~Xios();
+
+    void finalize();
+    void context_finalize();
+    bool isInitialized();
+
+    void configure() override;
+    void configureServer();
+    void configureCalendar();
+
+    cxios_date getCalendarOrigin();
+    cxios_date getCalendarStart();
+    xios::CDate getCurrentDate();
+    cxios_duration getCalendarTimestep();
+    void setCalendarOrigin(cxios_date origin);
+    void setCalendarStart(cxios_date start);
+    void setCalendarTimestep(cxios_duration timestep);
+
+    void getCalendarConfiguration();
+
+    std::string getCalendarDate(bool isoFormat = true);
+    int getCalendarStep();
+
+    void updateCalendar(int stepNumber);
+    void write(const std::string fieldstr, double* data, const int ni, const int nj);
+
+    std::string convertXiosDatetimeToString(cxios_date datetime, bool isoFormat = true);
+
+    void printCXiosDate(cxios_date date);
+    void printCXiosDuration(cxios_duration duration);
+
+    enum {
+        ENABLED_KEY,
+    };
+
+    int rank { 0 };
+    int size { 0 };
```

Comment:
I wonder if these should have clearer names - it's not obvious that the size of the XIOS handler would be the size of the associated MPI communicator.

Reply:
Feel free to update these when you implement your changes to the class.